src/filters/body.rs

```diff
@@ -255,6 +259,43 @@ pub fn form<T: DeserializeOwned + Send>() -> impl Filter<Extract = (T,), Error =
         .and_then(from_reader)
 }
 
+/// Returns a `Filter` that matches any request and extracts a `Future` of a protobuf encoded body.
+///
+/// # Note
+///
+/// # Warning
+///
+/// This does not have a default size limit, it would be wise to use one to
+/// prevent an overly large request from using too much memory.
+///
+///
+/// ```
+/// // Generated by PROST!
+/// #[derive(Clone, PartialEq, ::prost::Message)]
+/// pub struct UserRequest {
+///     #[prost(string, tag = "1")]
+///     pub name: std::string::String,
+/// }
+///
+/// let route = warp::body::content_length_limit(1024 * 16)
+///             .and(warp::body::proto())
+///             .map(|user: HelloUser| {
+///                 format!("Hello {}!\nUserID: {}", user.name, user.id)
+///             });
+/// ```
+#[cfg(feature = "protobuf")]
+pub fn protobuf<T: Message + Send + Default>(
+) -> impl Filter<Extract = (T,), Error = Rejection> + Copy {
+    async fn from_reader<T: Message + Send + Default>(buf: impl Buf) -> Result<T, Rejection> {
+        T::decode(buf).map_err(|err| {
+            log::debug!("request protobuf body error: {}", err);
+            reject::known(BodyDeserializeError { cause: err.into() })
+        })
+    }
+
+    aggregate().and_then(from_reader)
```

Comment:
noteably, this does not include a content-type check, like the `form(...)` or `json(...)`, that is because the mime crate does not have one. I plan to submit a PR to that crate to add the MIME type, but have yet to